src/block-cursor.ts

```diff
@@ -130,28 +137,38 @@ function getBase(view: EditorView) {
 }
 
 function measureCursor(cm: CodeMirror, view: EditorView, cursor: SelectionRange, primary: boolean): Piece | null {
-  let head = cursor.head
-  var fatCursor = false
-  var hCoeff = 1;
-  var vim = cm.state.vim;
+  let head = cursor.head;
+  let fatCursor = false;
+  let hCoeff = 1;
+  let vim = cm.state.vim;
   if (vim && (!vim.insertMode || cm.state.overwrite)) {
-    fatCursor = true
+    fatCursor = true;
     if (vim.visualBlock && !primary)
       return null;
     if (cursor.anchor < cursor.head) head--;
-    if (cm.state.overwrite) hCoeff = 0.2
-    else if (vim.status) hCoeff = 0.5
+    if (cm.state.overwrite) hCoeff = 0.2;
+    else if (vim.status) hCoeff = 0.5;
   }
 
-  
-  let pos = view.coordsAtPos(head, cursor.assoc || 1)
-  if (!pos) return null
-  let base = getBase(view)
   if (fatCursor) {
-    let letter = head < view.state.doc.length && view.state.sliceDoc(head, head + 1) 
-    if (!letter || letter == "\n"  || letter == "\r") letter = "\xa0"
-    let h =  (pos.bottom - pos.top) 
-    return new Piece(pos.left - base.left, pos.top - base.top + h * (1-hCoeff), h * hCoeff,
+    let pos = view.coordsAtPos(head, cursor.assoc || 1);
+    if (!pos) return null;
+    let base = getBase(view);
+    let domAtPos = view.domAtPos(head);
+    let node = domAtPos ? domAtPos.node : view.contentDOM;
+    while (domAtPos && domAtPos.node instanceof HTMLElement) {
```

Comment:
Should this be if, as after one iteration offset is undefined.
Also i think codemirror returns offset for childNodes not children.

Reply:
Oops!

Reply:
I'm using `while` because if you have multiple view plugins adding mark styles then it will result in multiple layers of nested spans.